build-support/bin/generate_docs.py

```diff
@@ -42,6 +43,10 @@
 def main() -> None:
     logging.basicConfig(format="[%(levelname)s]: %(message)s", level=logging.INFO)
     args = create_parser().parse_args()
+
+    if args.sync and not args.api_key:
```

Comment:
It was bugging me that we didn't validate this, and you'd get a much later error.